.buildkite/metricbeat/pipeline.yml

```diff
@@ -1,5 +1,114 @@
 
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+
+  - group: "Mandatory Tests"
+    key: "mandatory-tests"
+    steps:
+      - label: ":linux: Ubuntu Unit Tests"
+        key: "mandatory-linux-unit-test"
+        command: ".buildkite/scripts/unit_tests.sh ${PIPELINE_NAME}"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "c2-standard-16"
+        artifact_paths: "metricbeat/build/*.*"
+
+      - label: ":go: Go Intergration Tests"
+        key: "mandatory-int-test"
+        command: ".buildkite/scripts/go_int_tests.sh ${PIPELINE_NAME}"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "c2-standard-16"
+        artifact_paths: "metricbeat/build/*.*"
+
+      - label: ":python: Python Integration Tests"
+        key: "mandatory-python-int-test"
+        command: ".buildkite/scripts/py_int_tests.sh ${PIPELINE_NAME}"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "c2-standard-16"
+        artifact_paths: "metricbeat/build/*.*"
+
+      - label: ":negative_squared_cross_mark: Cross compile"
+        key: "mandatory-cross-compile"
+        command: ".buildkite/scripts/crosscompile.sh ${PIPELINE_NAME}"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "c2-standard-16"
+        artifact_paths: "metricbeat/build/*.*"
+
+      - label: ":windows: Windows Unit Tests - {{matrix.image}}"
+        command: ".buildkite/scripts/win_unit_tests.ps1"
+        key: "mandatory-win-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "{{matrix.image}}"
+          machine_type: "n2-standard-8"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        matrix:
+          setup:
+            image:
+              - "${IMAGE_WIN_2016}"
+              - "${IMAGE_WIN_2022}"
+        artifact_paths: "metricbeat/build/*.*"
+
+  - group: "Extended Tests"
+    key: "extended-tests"
+    steps:
+      - label: ":mac: MacOS Unit Tests"
+        key: "extended-macos-unit-tests"
+        command: ".buildkite/scripts/unit_tests.sh ${PIPELINE_NAME}"
+        agents:
+          provider: "orka"
+          imagePrefix: "${IMAGE_MACOS_X86_64}"
+        artifact_paths: "metricbeat/build/*.*"
+
+  - group: "Extended Windowds Tests"
+    key: "extended-win-tests"
+    steps:
+      - label: ":windows: Win 2019 Unit Tests"
+        key: "extended-win-2019-unit-tests"
+        command: ".buildkite/scripts/win_unit_tests.ps1"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machine_type: "n2-standard-8"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths: "metricbeat/build/*.*"
+
+  - group: "Packaging"
+    key: "packaging"
+    if: build.env("BUILDKITE_TAG") == "" && build.env("BUILDKITE_PULL_REQUEST") != ""
```

Comment:
I'm certainly confused with this condition.

IMO, the packaging should run for the below conditions:
1) By default on merge commits in any of the active branches (main, 7.17 8.12, ...)
2) On a PR basis, if changes in the changesets as described in https://github.com/elastic/beats/pull/37592/files#diff-097d0bb101a7c06cc829db46747adc87108f6a3cc5466b4c49be6566477799c5R9-R16
3) On a PR basis, if a particular GitHub comment or a tag, no strong opinion whether both options or just one... but that's how people can create packages on their PRs to do other things if needed, even if the above-mentioned conditions are false.

I think the `mandatory-tests` group should always be executed before the packaging, but `"extended-tests"` or `"extended-win-tests"` should not block the `packaging` group from being executed.

Reply:
This condition was gotten from the [Jenkins](https://github.com/elastic/beats/blob/0dc012db7247246a1439311686e7a7cb8515efdb/metricbeat/Jenkinsfile.yml#L97-L103):

Reply:
Jenkins implementation uses two different approaches:
- either running the packaging in the main branch
- or running the downstream packaging pipeline

https://github.com/elastic/beats/blob/0dc012db7247246a1439311686e7a7cb8515efdb/Jenkinsfile#L160-L195

I don't recall quite well what was the reason for that particular implementation, but https://github.com/elastic/beats/pull/28589 explained a bit the rationale.

In any case, that's a Jenkins specific implementation, but `packaging` should always run for branches and on a PR basis if changes in some files or explicitly added a GitHub comment.